contracts/libraries/OracleLibrary.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+
+import '@uniswap/v3-core/contracts/libraries/FullMath.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/LowGasSafeMath.sol';
+
+/// @title Oracle library
+/// @notice Provides functions to compute time weighted token conversion rate using the oracle
+library OracleLibrary {
+    uint32 constant UINT32_MAX = uint32(-1);
+
+    /// @notice Calculate time weighted token conversion rate at a time range from a single pool
+    /// @dev This function supports time range that spans across uint32 overflow boundaries
+    /// @param currentBlockTimestamp The current block timestamp
+    /// @param poolAddress The address of the pool to observe
+    /// @param tokenIn The address of the ERC20 token contract of amountIn
+    /// @param amountIn The token input amount before conversion
+    /// @param startBlockTimestamp The start of the time range to be observed
+    /// @param endBlockTimestamp The end of the time range to be observed
+    /// @return amountOut The token output amount after conversion
+    function consult(
+        uint32 currentBlockTimestamp,
+        address poolAddress,
+        address tokenIn,
```

Comment:
there's nothing going in, so i'd prefer using the names `base` and `quote` instead of `in` and `out`